services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/AttachmentServiceImpl.kt

```diff
@@ -0,0 +1,110 @@
+package fi.hel.haitaton.hanke.attachment
+
+import fi.hel.haitaton.hanke.HankeNotFoundException
+import fi.hel.haitaton.hanke.HankeRepository
+import fi.hel.haitaton.hanke.currentUserId
+import java.time.OffsetDateTime
+import java.util.UUID
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.web.multipart.MultipartFile
+
+private val supportedFiletypes =
+    mapOf(
+        "application/pdf" to setOf("pdf"),
+        "application/msword" to setOf("docx", "doc"),
+        "image/jpeg" to setOf("jpg", "jpeg"),
+        "image/png" to setOf("png"),
+        "image/vnd.dwg" to setOf("dwg", "dws")
+    )
+
+private const val FILESIZE_MAXIMUM_MEGABYTES = 10
+
+class AttachmentServiceImpl(
+    val hankeRepository: HankeRepository,
+    val hankeAttachmentRepository: HankeAttachmentRepository,
+) : AttachmentService {
+
+    fun getExtension(filename: String): String {
```

Comment:
This can be replaced at the call site with FilenameUtils.getExtension(filename) from Apache Commons.

Reply:
Fixed.